src/store.ts

```diff
@@ -1,10 +1,16 @@
-import { configureStore } from '@reduxjs/toolkit';
+import { combineReducers, configureStore } from '@reduxjs/toolkit';
+import productListReducer from '@reducers/listSlice';
 import { ecommerceApi } from '@services/ecommerce';
 
+export type RootState = ReturnType<typeof rootReducer>;
+
+const rootReducer = combineReducers({
+  [ecommerceApi.reducerPath]: ecommerceApi.reducer,
+  productList: productListReducer,
+});
+
 export const store = configureStore({
-  reducer: {
-    [ecommerceApi.reducerPath]: ecommerceApi.reducer,
-  },
+  reducer: rootReducer,
   middleware: (getDefaultMiddleware) =>
     getDefaultMiddleware().concat(ecommerceApi.middleware),
```

Comment:
better practice to use modern syntax [...getDefaultMiddleware(), ecommerceApi.middleware], coz when you will have 10 or more middlewares you will see concat.****.concat.*****.concat

Reply:
okay